types/rlp_unmarshal.go

```diff
@@ -396,8 +396,10 @@ func (t *Transaction) unmarshalRLPFrom(p *fastrlp.Parser, v *fastrlp.Value) erro
 	switch t.Type {
 	case LegacyTx:
 		num = 9
-	case DynamicFeeTx, StateTx:
+	case StateTx:
 		num = 10
+	case DynamicFeeTx:
+		num = 12
```

Comment:
Because it has chain ID and access list fields. We don't use them but should be compatible.